src/sciline/domain.py

```diff
@@ -3,36 +3,55 @@
 from typing import Any, Generic, TypeVar, get_args, get_origin
 
 PARAM = TypeVar("PARAM")
+PARAM2 = TypeVar("PARAM2")
 SUPER = TypeVar("SUPER")
 
 
+def _check_supertype(cls: type, scope_cls: type) -> None:
+    # Mypy does not support __orig_bases__ yet(?)
+    # See also https://stackoverflow.com/a/73746554 for useful info
+    scope = cls.__orig_bases__[0]  # type: ignore[attr-defined]
+    # Only check direct subclasses
+    if get_origin(scope) is scope_cls:
+        supertype = get_args(scope)[-1]
+        # Remove potential generic params
+        # In Python 3.8, get_origin does not work with numpy.typing.NDArray,
+        # but it defines __origin__
+        supertype = getattr(supertype, '__origin__', None) or supertype
```

Comment:
Is it assuming any of `Scope` classes does not have `__origin__` defined...?

Reply:
No sorry I was a bit confused... why is it getting `__origin__` of the `supertype`...?
Can you tell me a use-case...? 🥲

Reply:
If you make an alias of a generic type, this is needed for the `__bases__` check below.

Note that this is just moved, i.e., not new in this PR.

Reply:
Ah right now I understood. Thank you!